pkg/controller/build/build_controller_test.go

```diff
@@ -156,6 +156,72 @@ func TestBuildControllerMultipleOptedInPools(t *testing.T) {
 	})
 }
 
+// Tests that if multiple goroutines attempt to create or update a
+// canonicalized secret that all requests are successful.
+func TestCanonicalizedSecrets(t *testing.T) {
+	t.Parallel()
+
+	clients := getClientsForTest()
+	bc := NewWithCustomPodBuilder(BuildControllerConfig{MaxRetries: 1, UpdateDelay: testUpdateDelay}, clients)
+
+	legacyPullSecretName := "final-image-push-secret"
+
+	assertIsCanonicalSecret := func(t *testing.T, s *corev1.Secret) {
+		// We expect the returned secret to contain the canonical suffix, to have
+		// the canonical labels, and to generally be recognized as a canonical
+		// secret.
+		assert.Contains(t, s.Name, canonicalSecretSuffix)
+		assert.True(t, isCanonicalizedSecret(s))
+		assert.True(t, hasCanonicalizedSecretLabels(s))
```

Comment:
Doesn't isCanonicalizedSecret(s) already have a call within it for hasCanonicalizedSecretLabels(s)?